core/include/moveit/task_constructor/stages/property_passthrough.h

```diff
@@ -0,0 +1,60 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2017, Bielefeld University
+ *  Copyright (c) 2017, Hamburg University
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Bielefeld University nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+/* Authors: Captain Yoshi */
+
+#pragma once
+
+#include <moveit/task_constructor/stage.h>
+
+namespace moveit {
+namespace task_constructor {
+namespace stages {
+
+/** Generic passthrough Propagator for storing user defined properties
+ *
+ * Useful for post-planning purposes
+ */
+
+class PropertyPassThrough : public PropagatingEitherWay
+{
+public:
+	PropertyPassThrough(const std::string& name = "property passthrough");
+
+	void computeForward(const InterfaceState& from) override;
+	void computeBackward(const InterfaceState& to) override;
```

Comment:
I suggest to override `bool compute(const InterfaceState& /*state*/, planning_scene::PlanningScenePtr& /*scene*/, SubTrajectory& /*trajectory*/, Interface::Direction /*dir*/)` instead of implementing the same thing twice.

Reply:
Done. Shouldn't this method be protected instead of private in the [PropagatingEitherWay](https://github.com/ros-planning/moveit_task_constructor/blob/master/core/include/moveit/task_constructor/stage.h#L315) ?